fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DbFedoraToOcflObjectIndex.java

```diff
@@ -265,7 +266,11 @@ private void upsert(final String transactionId, final FedoraId fedoraId, final S
         parameterSource.addValue("ocflId", ocflId);
         parameterSource.addValue("transactionId", transactionId);
         parameterSource.addValue("operation", operation);
-        jdbcTemplate.update(UPSERT_MAPPING_TX_MAP.get(dbPlatform), parameterSource);
+        try {
+            jdbcTemplate.update(UPSERT_MAPPING_TX_MAP.get(dbPlatform), parameterSource);
+        } catch (Exception e) {
```

Comment:
This logic is not specific enough. It's possible for the `update` call to fail for reason unrelated to the length of the id.

Reply:
so, should I check the length of the id, rather than just wrap it in an exception handler?

Reply:
Checking the length may be reasonable, though, if we go that route, I think it would make more sense if the id length enforcement happened up in the HTTP layer.

Catching the exception here is not a problem. You just need to be more targeted about it. What is the type of exception that's thrown when the id is too long? What is the message of the exception? Are these consistent across database implementations? Using that information, you should be able to inspect the exception and only throw the id length exception when appropriate.